velox/functions/sparksql/String.h

```diff
@@ -455,4 +455,139 @@ struct SubstrFunction {
   }
 };
 
+struct OverlayFunctionBase {
+  template <bool isAscii, bool isVarchar>
+  FOLLY_ALWAYS_INLINE void doCall(
+      exec::StringWriter<false>& result,
+      const StringView& input,
+      const StringView& replace,
+      const int32_t pos,
+      const int32_t len) {
+    // Calculate and append first part.
+    auto pair = substring<isAscii, isVarchar>(input, 1, pos - 1);
+    if constexpr (isVarchar) {
+      auto byteRange = stringCore::getByteRange<isAscii>(
+          input.data(), pair.first + 1, pair.second);
+      result.append(StringView(
+          input.data() + byteRange.first, byteRange.second - byteRange.first));
+    } else {
+      result.append(StringView(input.data() + pair.first, pair.second));
+    }
+
+    // Append second part.
+    result.append(replace);
+
+    // Calculate and append last part.
+    int32_t length = 0;
+    if constexpr (isVarchar) {
+      length = len >= 0 ? len : stringImpl::length<isAscii>(replace);
+    } else {
+      length = len >= 0 ? len : replace.size();
+    }
+    int64_t start = (int64_t)pos + (int64_t)length;
```

Comment:
Why casting to int64_t? Are you trying to avoid overflow?

Reply:
Yes, I found some overflow cases in fuzzer tests.